docs/coffee_machine_concept.md

```diff
@@ -0,0 +1,194 @@
+engineering to describe the functional requirements of a system, such as an
+espresso machine. In the context of the espresso machine, a use case diagram
+would be used to identify and define the different ways in which the machine
+will be used by its users, such as the café staff and customers.
+
+The diagram would typically include different actors or users, such as the
+barista, the customer, and possibly a manager or maintenance technician. It
+would also include different "use cases" or scenarios, which describe the
+different actions that the users can take with the machine, such as placing an
+order, making an espresso, or cleaning the machine.
+
+The use case diagram helps to ensure that all the necessary functional
+requirements of the espresso machine are identified and accounted for, and that
+the system is designed to meet the needs of its users. It can also be used as a
+communication tool between the different stakeholders involved in the
+development of the machine, such as the ants and Cappuccino the cat.
+
+The ants need your help updating the diagrams, so let's get started:
+
+1. Double-click on the actor to pop up the rename dialog, and replace User with
+   Barista.
+2. Update the name of the oval Use Case from Use Case #1 to Brew espresso.
+3. Update the name of the rectangle Block from Feature to Espresso Machine
+
+A barista interacts with the espresso machine.
+The barista is provided a simple interface with a few push buttons.
+
+In this particular use case diagram, we have one actor named Barista and one
+use case called Brew espresso, which is allocated to a block called Espresso
+Machine. The actor, in this case, is a cat barista who interacts with the
+system (an espresso machine) to accomplish a particular task, which is brewing
+espresso.
+
+```{diagram} Use Cases
+:model: coffee-machine
+:alt: Use case diagram showing an actor named Barista and a use case called Brew espresso
+```
+
+The use case Brew espresso represents a specific functionality or action that
+the system (the Espresso Machine block) can perform. It describes the steps or
+interactions necessary to complete the task of brewing espresso, such as
+selecting the appropriate settings, starting the brewing process, and stopping
+the process once it's complete.
+
+The use case diagram shows the relationship between the actor and the use case.
+It is represented by an oval shape with the use case name inside and an
+association with the actor. The association represents the interaction from the
+actor to the use case.
+
+## Domain Diagram
+
+A domain diagram is a graphical representation of the concepts, terms, and
+relationships within a specific domain. In the case of a coffee shop, a domain
+diagram could represent the key elements and relationships within the coffee
+shop domain.
+
+The following is a domain diagram that builds upon the context diagram with
+additional blocks:
+- Barista
```

Comment:
The Barista is an actor, right? Is it also a Block?

Reply:
Yes, although sometimes I have seen using a Block as the actor itself: https://mbse4u.com/2012/03/20/the-death-of-the-actor/

I'm hoping SysML v2 cleans this up a bit.